### ISSUE TITLE
Large Size PDF(pages) causes Browser Crash

### ISSUE THREAD

**Issue by SIlver--** [VadimDez/ng2-pdf-viewer#410]
##### Bug Report or Feature Request (mark with an `x`)
```
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [ X ] Bug report -> please search issues before submitting
- [ ] Feature request
- [ ] Documentation issue or request
```
Error in question is 
```
Warning: Unable to decode image: RangeError: Array buffer allocation failed
```
I've attached the file in question. The PDF has 600+ pages. Before all the pages get rendered it fails with the error. 
[A17_FlightPlan.pdf](https://github.com/VadimDez/ng2-pdf-viewer/files/2567115/A17_FlightPlan.pdf)

**Comment by liaothomas** [comment]
What I found is that there is a definite limit to the number of pages that can be accommodated with show all.  I have successfully shown a 4000+ page document in groups of about 10 pages at a time.  With a 600+ page document, I would expect that about 30 pages could be shown at once.  In chasing this down, the best indication is that the issue is with the amount of memory available to the browser session.    

I read in the pdfjs documentation (The underlying tool used by this package) that they acknowledge that they don't manage memory optimally.  It is on the to do list but there was no indication about when they might be addressing this issue.
I hope that this is useful information to help with the ultimate resolution of the document size issue.  A suggestion would be to enable the display of a range of pages rather than show all or a single page.  It will help with paging through a large document.

**Comment by SIlver--** [comment]
Thanks @liaothomas. Does `ng2-pdf-viewer` allow for a range of files? How do you go about dividing it out? 

I know `[page]` allows you to show a single page but the only way I've found to do a range in the past is with calling an incremental implementation of the `ng2-pdf-viewer` component.

**Comment by vishnu-dev** [comment]
> What I found is that there is a definite limit to the number of pages that can be accommodated with show all. I have successfully shown a 4000+ page document in groups of about 10 pages at a time. With a 600+ page document, I would expect that about 30 pages could be shown at once. In chasing this down, the best indication is that the issue is with the amount of memory available to the browser session.
> 
> I read in the pdfjs documentation (The underlying tool used by this package) that they acknowledge that they don't manage memory optimally. It is on the to do list but there was no indication about when they might be addressing this issue.
> I hope that this is useful information to help with the ultimate resolution of the document size issue. A suggestion would be to enable the display of a range of pages rather than show all or a single page. It will help with paging through a large document.

@liaothomas Could you please explain more on  **in groups of about 10 pages at a time** and how did you achieve loading 4000+ pages document.

My implementation craps out on 150+ pages itself. The loader uses 1.2GB of RAM to render 180page document with show-all=true

**Comment by varun-nagar** [comment]
Facing same issue, please revert with possible solution with show-all=true for large size pdf.

**Comment by vishnu-dev** [comment]
> > What I found is that there is a definite limit to the number of pages that can be accommodated with show all. I have successfully shown a 4000+ page document in groups of about 10 pages at a time. With a 600+ page document, I would expect that about 30 pages could be shown at once. In chasing this down, the best indication is that the issue is with the amount of memory available to the browser session.
> > I read in the pdfjs documentation (The underlying tool used by this package) that they acknowledge that they don't manage memory optimally. It is on the to do list but there was no indication about when they might be addressing this issue.
> > I hope that this is useful information to help with the ultimate resolution of the document size issue. A suggestion would be to enable the display of a range of pages rather than show all or a single page. It will help with paging through a large document.
> 
> @liaothomas Could you please explain more on **in groups of about 10 pages at a time** and how did you achieve loading 4000+ pages document.
> 
> My implementation craps out on 150+ pages itself. The loader uses 1.2GB of RAM to render 180page document with show-all=true

Any updates on this @VadimDez @liaothomas?

**Comment by VadimDez** [comment]
Fixed in `6.0.0` See https://github.com/VadimDez/ng2-pdf-viewer/blob/master/CHANGELOG.md#600
